MetaMask/metamask-extension
# Remove extraneous methods from NetworkController

## Explanation



The network controller has a variety of methods that just retrieve
controller state. These methods are not necessary because controller
state is already part of the public API of the controller and can be
accessed directly.

These methods are:

- getCurrentChainId
- getCurrentRpcUrl
- getNetworkIdentifier
- getNetworkState
- getProviderConfig
- isNetworkLoading

This is part of a larger effort to normalize the API of both network
controllers, to make them easier to merge.

Fixes #16947.

## Manual Testing Steps



The changes here touch the following pieces:

* ENS IPFS resolver
* MetaMetrics controller
* network provider
* token list controller
* gas fee controller
* ENS controller
* incoming transactions controller
* account tracker
* cached balances controller
* transaction controller
* currency rate controller
* typed message manager
* swaps controller
* smart transactions controller
* legacy provider config stream
* `metamask_getProviderState`
* `wallet_switchEthereumChain`
* `wallet_addEthereumChain`

So it appears that we should test:

* That you can't visit an IPFS or ENS URL while not on mainnet
* Anything involving metrics (should send the current chain ID along as data)
* General wallet functionality (which should trigger RPC requests through the provider)
* Adding tokens (should see new tokens as they arrive)
* Making a transaction (should see gas fees for current network, should see current token<->fiat conversion rate, should see new transaction in list after creation)
* Should see ENS names resolved for accounts
* Trigger a signature confirmation, then switch network (signing should fail)
* Swaps should work for current network
* Transaction insights should work for current network
* Data in legacy provider config stream should include current chain ID
  * This one might be difficult to test as I'm not sure how often this is used, or how many people know about this. I certainly didn't!
* Return data from `metamask_getProviderState` should include the current chain ID
* `wallet_switchEthereumChain` shouldn't do anything when the chain ID is the same as the current network
* `wallet_addEthereumChain` shouldn't do anything when the chain ID / RPC URL being added is the same as the current network

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
